chaos/actor.go

```diff
@@ -115,3 +117,19 @@ func (a Actor) CreateActor(rt runtime.Runtime, args *CreateActorArgs) *adt.Empty
 	rt.CreateActor(acid, addr)
 	return nil
 }
+
+type ResolveAddressResponse struct {
+	Address address.Address
+	Success bool
+}
+
+func (a Actor) ResolveAddress(rt runtime.Runtime, args *address.Address) *address.Address {
+	rt.ValidateImmediateCallerAcceptAny()
+	
+	resolvedAddr, ok := rt.ResolveAddress(*args)
+	if !ok {
+		invalidAddr, _ := address.NewIDAddress(0)
```

Comment:
Using the system actor's implicit address as a sentinel value for when resolution fails is hard to follow. How about we add a `ResolveAddressReturn` type that contains the address and a boolean indicating if resolution failed? It's a lot more explicit and self-descriptive. See how `CreateActor` uses this technique on the input side.

Reply:
the downsides of new types is that i would also need to either add cbor-gen machinery around it, as would any other implementations, right?

Reply:
Good news is that it's already there ;-) Add your type to: https://github.com/filecoin-project/test-vectors/blob/master/chaos/gen/gen.go, then execute `go generate`

https://github.com/filecoin-project/test-vectors/blob/a5d1c62f093bd44e487a35211031a5e5a0824936/chaos/actor.go#L13